dcpy/lifecycle/package/abstract_doc.py

```diff
@@ -0,0 +1,360 @@
+                                style=de.CellStyle(
+                                    font=de.Font(bold=True, size=11),
+                                ),
+                            ),
+                            de.Cell(
+                                value=(r["description"]),
+                                style=de.CellStyle(
+                                    font=de.Font(size=9, italic=True),
+                                ),
+                            ),
+                        ],
+                    ),
+                    de.Cell(  # Value
+                        value=value,
+                        style=de.CellStyle(font=de.Font(italic=True)),
+                    ),
+                ]
+            )
+        )
+
+    return de.Table(
+        title=title,
+        subtitle=subtitle,
+        description=description,
+        rows=rows,
+        column_widths=column_widths or [50, 80],
+    )
+
+
+def make_list_table(
+    title: str,
+    subtitle: str,
+    table_rows: list[list[Any]],
+    column_headers: list[str] = [],
+    column_ids: list[str] = [],
+    column_descriptions: list[str] = [],
+    description: str | None = None,
+    maybe_image_path: Path | None = None,
+    column_widths: list[float] | None = None,
+    style_overrides: dict[str, de.CellStyle] = {},
+):
+    """Make a table for a list of objects. e.g. Revisions or columns
+    There is one row per column, and columns can be specified
+    """
+    rows = _make_table_top(
+        title=title,
+        subtitle=subtitle,
+        description=description,
+        maybe_image_path=maybe_image_path,
+        column_headers=column_headers,
+        column_header_descriptions=column_descriptions,
+    )
+
+    for tr in table_rows:
+        cells = []
+        for val, col_id in zip(tr, column_ids):
+            style = style_overrides.get(col_id, de.CellStyle())
+            style.text_alignment_vertical = style.text_alignment_vertical or "top"
+            cells.append(de.Cell(value=val, style=style))
+        rows.append(de.Row(cells=cells))
+
+    return de.Table(
+        title=title,
+        subtitle=subtitle,
+        description=description,
+        rows=rows,
+        column_widths=column_widths or [50, 80],
+    )
+
+
+# TODO: move to pydantic models
+def get_field_metadata(data_source: str, org_metadata: OrgMetadata):
+    match data_source:
+        case "dataset.columns":
+            dictionary_section = org_metadata.data_dictionary.dataset["columns"]
+        case "dataset.revisions":
+            dictionary_section = org_metadata.data_dictionary.dataset["revisions"]
+        case "dataset.attributes":
+            dictionary_section = org_metadata.data_dictionary.dataset["attributes"]
+        case _:
+            raise Exception(f"Unknown data_source: {data_source}")
+    return dictionary_section
+
+
+# TODO: move to pydantic models
+def get_data_source(
```

Comment:
@fvankrieken and this would be folded into the `tabular_report`, which could probably be a Pydantic class from which we'd inherit. (would probably move `get_field_metadata` into that class as well)